securesystemslib/signer/_signer.py

```diff
@@ -0,0 +1,264 @@
+
+        Returns a specific Signer instance based on the private key URI and the
+        supported uri schemes listed in SIGNER_FOR_URI_SCHEME.
+
+        Args:
+            priv_key_uri: URI that identifies the private key
+            public_key: Key that is the public portion of this private key
+            secrets_handler: Optional function that may be called if the
+                signer needs additional secrets (like a PIN or passphrase).
+                secrets_handler should return the requested secret string.
+
+        Raises:
+            ValueError: Incorrect arguments
+            Other Signer-specific errors: These could include OSErrors for
+                reading files or network errors for connecting to a KMS.
+        """
+
+        scheme, _, _ = priv_key_uri.partition(":")
+        if scheme not in SIGNER_FOR_URI_SCHEME:
+            raise ValueError(f"Unsupported private key scheme {scheme}")
+
+        signer = SIGNER_FOR_URI_SCHEME[scheme]
+        return signer.from_priv_key_uri(
+            priv_key_uri, public_key, secrets_handler
+        )
+
+
+class SSlibSigner(Signer):
+    """A securesystemslib signer implementation.
+
+    Provides a sign method to generate a cryptographic signature with a
+    securesystemslib-style rsa, ed25519 or ecdsa key. See keys module
+    for the supported types, schemes and hash algorithms.
+
+    SSlibSigners should be instantiated with Signer.from_priv_key_uri().
+    These private key URI schemes are supported:
+    * "envvar:<VAR>":
+        VAR is an environment variable with unencrypted private key content.
+           envvar:MYPRIVKEY
+    * "file:<PATH>?encrypted=[true|false]":
+        PATH is a file path to a file with unencrypted private key content.
+           file:path/to/file?encrypted=true
+           file:/abs/path/to/file?encrypted=false
+
+    Attributes:
+        key_dict:
+            A securesystemslib-style key dictionary. This is an implementation
+            detail, not part of public API
+    """
+
+    ENVVAR_URI_SCHEME = "envvar"
+    FILE_URI_SCHEME = "file"
+
+    def __init__(self, key_dict: Dict):
+        self.key_dict = key_dict
+
+    @classmethod
+    def from_priv_key_uri(
+        cls,
+        priv_key_uri: str,
+        public_key: Key,
+        secrets_handler: Optional[SecretsHandler] = None,
+    ) -> "SSlibSigner":
+        """Constructor for Signer to call
+
+        Arguments:
+            priv_key_uri: private key URI described in class doc
+            public_key: Key object.
+
+        Raises:
+            OSError: Reading the file failed with "file:" URI
+            ValueError: URI is unsupported or environment variable was not set
+                with "envvar:" URIs
+
+        Returns:
+            SSlibSigner for the given private key URI.
+        """
+        if not isinstance(public_key, SSlibKey):
+            raise ValueError(f"Expected SSlibKey for {priv_key_uri}")
```

Comment:
I decided not to:
* Signer.from_priv_key_uri() already states that it  rasies `ValueError: Incorrect arguments` and that the given public key must be for this specific private key
* I think it's reasonable to expect that users will only read Signer docs (since that's the class they use) and not the implementation docs

There is a related issue here: I left a longer comment in reply to martin about exceptions from signer implementations...